### PR TITLE
Tools: Improve configuration for formatting files

### PR DESCRIPTION
**Pull request by gziolo**
## What?

I run into some issues with YAML file formatting when working on #40976 because my IDE was entering tabs instead of spaces. I also noticed that we don't run formatting on JSON files despite having all of them formatted with Prettier in the past.

## Why?

It turns out that YAML doesn't allow tabs, see https://yaml.org/faq.html:

> Tabs have been outlawed since they are treated differently by different editors and tools. And since indentation is so critical to proper interpretation of YAML, this issue is just too tricky to even attempt. Indeed Guido van Rossum of Python has acknowledged that allowing TABs in Python source is a headache for many people and that were he to design Python again, he would forbid them.

I figured out I contributed to that when I removed overrides for YAML in `.editorconfig` files in https://github.com/WordPress/gutenberg/pull/30794 😓 

I also discovered that `@wordpress/scripts` formats YAML but not JSON. So I thought we should add it to the list since we formatted all JSON files with Prettier in the past in https://github.com/WordPress/gutenberg/pull/30714.

## How?

Enables code formatting for all JSON files in `@wordpress/scripts` and the `format` command.

I executed `npm run format` for the Gutenberg repository to ensure that formatting for JSON and YAML files is fixed.

I updated the pre-commit hook to ensure that formatting triggers correctly for JSON and YAML files. I updated `lint-staged` to v10 which by default automatically adds modified (formatted) files to the current commit.

## Testing Instructions

1. Run `npm run format` and make sure there are no changes applied.
2. Type a tab in the YAML file and confirm that your IDE inserts two spaces instead.
3. Apply some random changes to the JSON file and confirm that `npm run format` fixes them.

### CODE REVIEW

File: packages/url/src/test/fixtures/wpt-data.json
Comment (ntwb) [inline packages/url/src/test/fixtures/wpt-data.json]: This doesn't look right... Not sure why, searching for `oof` in my browser returns 0 results...

Same no search results in the raw diff view too https://patch-diff.githubusercontent.com/raw/WordPress/gutenberg/pull/40994.diff

**Review by ntwb — CHANGES_REQUESTED** [review]
Nice, just 1 comment,maybe it's unicode, unsure

File: packages/url/src/test/fixtures/wpt-data.json
Comment (ntwb) [reply packages/url/src/test/fixtures/wpt-data.json]: Ha, yes, unicode, today I learned:

image

Comment (gziolo) [reply packages/url/src/test/fixtures/wpt-data.json]: I will exclude this file from formatting since it’s autogenerated with a script anyway 👍

File: package.json
Comment (gziolo) [inline package.json]: Does anyone know how to prevent commit when formatting applies changes to modified files?

File: packages/url/src/test/fixtures/wpt-data.json
Comment (ntwb) [reply packages/url/src/test/fixtures/wpt-data.json]: Yeah, the underlined section of this code is the affected line, unicode is weird, not sure how this get's `oof` added here /shrug
image

File: package.json
Comment (gziolo) [reply package.json]: It looks like `lint-staged` v10 adds changes to the commit automatically 🎉

File: packages/url/src/test/fixtures/wpt-data.json
Comment (gziolo) [reply packages/url/src/test/fixtures/wpt-data.json]: I added this file to the list of files that won't get formatted by Prettier and removed from this PR.

**Comment by gziolo** [comment]
I think it's ready for final review now.

**Review by dmsnell — APPROVED** [review]
No need for review from me so I'm lending my approval (please defer to others), nothing here looks risky to the parts I work in. Thanks for normalizing the formatting.

**Review by ntwb — APPROVED** [review]
Looks good 👌🏼

**Pull request merged**
